tests/test_controlled_policy_analysis.py

```
@@ -0,0 +1,181 @@
+    request = urllib.request.Request("https://example.com/privacy")
+
+    try:
+        handler.redirect_request(
+            request,
+            None,
+            302,
+            "Found",
+            {},
+            "https://vendor.example.net/privacy",
+        )
+        assert False, "Expected cross-host redirect to be blocked"
```

Ruff (0.16.2)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)